### PR TITLE
DROP COLUMN updates

### LINKED ISSUE #8142: sql: drop indexes that reference dropped columns
PR: https://github.com/cockroachdb/cockroach/pull/51661

From release notes:
> Dropping a column always drops all indexes that index the column. It is no longer necessary to provide the `CASCADE` option to [`DROP COLUMN`](../v20.2/drop-column.html). [#51661][#51661] {% comment %}doc{% endcomment %}

### PR DESCRIPTION
**Pull request by ericharmeling**
Fixes #8142.

### CODE REVIEW

File: v20.2/drop-column.md [hunk 1]
```diff
@@ -22,7 +26,7 @@ The user must have the `CREATE` [privilege](authorization.html#assign-privileges
 -----------|-------------
  `table_name` | The name of the table with the column you want to drop.
  `name` | The name of the column you want to drop.<br><br>When a column with a `CHECK` constraint is dropped, the `CHECK` constraint is also dropped.
- `CASCADE` | Drop the column even if objects (such as [views](views.html)) depend on it; drop the dependent objects, as well.<br><br>`CASCADE` does not list objects it drops, so should be used cautiously. However, `CASCADE` will not drop dependent indexes; you must use [`DROP INDEX`](drop-index.html).<br><br>`CASCADE` will drop a column with a foreign key constraint if it is the only column in the reference.
+ `CASCADE` | Drop the column even if objects (such as [views](views.html)) depend on it; drop the dependent objects, as well. `CASCADE` will drop a column with a foreign key constraint if it is the only column in the reference.<br><br>`CASCADE` does not list the objects it drops, so should be used cautiously.<br><br><span class="version-tag">New in v20.2:</span> `CASCADE` is not required to drop an indexed column. `DROP COLUMN` drops the column and any indexes on the column by default.
```
Comment (mgartner) [inline v20.2/drop-column.md:29]: [nit] `DROP COLUMN` will also drop a partial index with a predicate that references the column being dropped. For example, `ALTER TABLE t DROP COLUMN c` would drop an `INDEX ON t (b) WHERE c > 0`.

Comment (mgartner) [reply v20.2/drop-column.md:29]: `ALTER TABLE t DROP COLUMN c` would also drop an `INDEX ON t (b) STORING (c)`.

**Review by mgartner — APPROVED** [review]
LGTM

**Review by ericharmeling — COMMENTED** [review]
TFTR @mgartner !

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/8660#-:-MJh5Ijv04r3p9GfaDcf:bcaz2iw)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @mgartner)

___
*[v20.2/drop-column.md, line 29 at r1](https://reviewable.io/reviews/cockroachdb/docs/8660#-MJbdDBwapd0Fz7XLMLP-r1-29:-MJh50AS3z42CpHXcYrP:bnsyfsj) ([raw file](https://github.com/cockroachdb/docs/blob/afeb9290835e360df4c549d2fc7391134bf76778/v20.2/drop-column.md#L29)):*
<details><summary><i>Previously, mgartner (Marcus Gartner) wrote…</i></summary><blockquote>

`ALTER TABLE t DROP COLUMN c` would also drop an `INDEX ON t (b) STORING (c)`.
</blockquote></details>

Thanks for pointing these out too! Updated this section, and the note at the top of the page.

**Review by mgartner — APPROVED** [review]
LGTM

**Review by lnhsingh — COMMENTED** [review]
LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/8660#-:-MJi1w1H1zTJshivzpcQ:bb6ywmx)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @lnhsingh and @mgartner)

**Pull request merged**